management/src/commands/update-identity.ts

```diff
@@ -0,0 +1,92 @@
+import { IdentityNotFoundError, InvalidValidationCodeError } from "../errors";
+import { checkVerificationCode } from "../queries/check-verification-code";
+import { getIdentity } from "../queries/get-identity";
+import { IdentityTypes, ManagementCredentials } from "../types";
+import { addIdentityToUser } from "./add-identity-to-user";
+import { markIdentityAsPrimary } from "./mark-identity-as-primary";
+import { removeIdentityFromUser } from "./remove-identity-from-user";
+
+type VerifyValidationCodeData = {
+  validationCode: string;
+  eventId: string;
+};
+
+type NewIdentityData = {
+  value: string;
+  type: IdentityTypes;
+  primary: string;
+};
+
+async function updateIdentity(
+  managementCredentials: ManagementCredentials,
+  userId: string,
+  verifyValidationCode: VerifyValidationCodeData,
+  newIdentity: NewIdentityData,
+  identityToUpdateId: string,
+): Promise<void> {
+  const { validationCode, eventId } = verifyValidationCode;
+  const { value, type, primary } = newIdentity;
+
+  const identityToUpdate = await getIdentity(managementCredentials, {
+    userId,
+    identityId: identityToUpdateId,
+  });
+
+  if (!identityToUpdate) {
+    throw new IdentityNotFoundError();
+  }
+
+  const { status: verificationStatus } = await checkVerificationCode(
+    managementCredentials,
+    {
+      code: validationCode,
+      eventId,
+    },
+  );
+
+  if (verificationStatus !== "VALID") {
+    throw new InvalidValidationCodeError();
+  }
+
+  const { id: identityId } = await addIdentityToUser(managementCredentials, {
+    userId,
+    identityType: type,
+    identityValue: value,
+  });
+
+  if (primary) {
+    await markIdentityAsPrimary(managementCredentials, identityId).catch(
+      async (error) => {
+        const identity = {
+          userId,
+          identityType: type,
+          identityValue: identityToUpdate.value,
+        };
```

Comment:
Shouldn't it be the new Identity value here ? Because, if I understand everything well, with this code, if we failed on marking the new one as primary, we remove the "old" one.

Reply:
👀